lib/matplotlib/tests/test_figure.py

```diff
@@ -553,6 +553,17 @@ def test_savefig_preserve_layout_engine(tmp_path):
     assert fig.get_layout_engine()._compress
 
 
+def test_savefig_locate_colorbar(tmp_path):
+    fig, ax = plt.subplots()
+    pc = ax.pcolormesh(np.random.randn(2, 2))
+    cbar = fig.colorbar(pc, aspect=40)
+    fig.savefig(tmp_path / 'foo.png', bbox_inches=mpl.transforms.Bbox([[0, 0], [4, 4]]))
+
+    # Check that an aspect ratio has been applied.
```

Comment:
Are you able to assert against the aspect ratio too?

Reply:
I couldn't figure out how to check the aspect ratio directly.  `get_aspect` returns "auto" before and after save.  `get_box_aspect` returns 40 before and after save.  Is there something else I've missed?

Reply:
I didn't know either, I just figured since it was the aspect that was wrong that is better to test against, so figured I would ask if you tried :) but colorbars are never straightforward and there is still an open issue with the get/set aspect on them. https://github.com/matplotlib/matplotlib/pull/22103

Although here it could also be that the aspect is only changed in the setattr context manager while saving the figure, so you don't get it except while in that context manager?

This still looks good to me!